crates/proto/src/h2/h2_client_stream.rs

```diff
@@ -722,15 +725,20 @@ mod tests {
         }
     }
 
-    #[test]
+    #[tokio::test]
     #[ignore = "cloudflare has been unreliable as a public test service"]
-    fn test_https_cloudflare() {
+    async fn test_https_cloudflare() {
         subscribe();
 
         let cloudflare = SocketAddr::from(([1, 1, 1, 1], 443));
         let mut request = Message::new();
         let query = Query::query(Name::from_str("www.example.com.").unwrap(), RecordType::A);
         request.add_query(query);
+        request.set_recursion_desired(true);
```

Comment:
Is this new? If necessary, maybe split it out in a separate commit from the one doing all the mechanical `tokio::test` changes?